src/modules/battery_status/battery_status.cpp

```diff
@@ -190,34 +183,40 @@ BatteryStatus::adc_poll()
 
 	int selected_source = -1;
 
-	if (ret >= (int)sizeof(buf_adc[0])) {
+	adc_report_s adc_report;
 
-		/* Read add channels we got */
-		for (unsigned i = 0; i < ret / sizeof(buf_adc[0]); i++) {
-			{
-				for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
+	if (_adc_report_sub.update(&adc_report)) {
 
-					/* Once we have subscriptions, Do this once for the lowest (highest priority
-					 * supply on power controller) that is valid.
+		/* Read add channels we got */
+		for (int i = 0; i < PX4_MAX_ADC_CHANNELS; ++i) {
+			for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
```

Comment:
The '{' was scoping, and it was removed.

Reply:
Oh sorry I missed it

Reply:
No. It was in original code but not deleted